sourcegraph/about
# Icon Styling

This closes #5432. It centralizes our icons in a flexible component and updates styling across the About site.

### Changelog

- Moves icons into `Icon` component
  - Accepts icon name, size, onClick, additional classes, etc.
  - Implements [new icon styling](https://www.figma.com/file/o1QRtdQI0ozKq0n7ATrKlx/Marketing-DLS?node-id=4422%3A86874) on `variant="boxed"`
  - Allows for plain icon when `variant` prop is omitted
- Uses the [Material UI icon library](https://www.npmjs.com/package/@mui/icons-material) identified in conversation with Tracey

New icon style:
<img width="600" alt="Screen Shot 2022-06-16 at 9 01 55 AM" src="https://user-images.githubusercontent.com/60713139/174087303-82884b88-0e97-4930-856f-28f0e634568e.png">

### Notes
- Spotify and Twitch logos are the only remaining mdi-react icons. They were [not available in the library](https://mui.com/material-ui/material-icons/?theme=Sharp). Are we open to replacing these with SVGs?
- I ran into an issue rendering blogposts with `BlockquoteWithBorder` due to the inclusion of the `Icon` component. TIL when a component is passed into another component, and then used as in MDX, Next does not use the Babel-aliased path when reading the component. So, `Icon` needs to be imported with a relative path in `BlockquoteWithBorder` for MDX to find it.
- Still working on optimizing the bundle size (see [comments](https://github.com/sourcegraph/about/issues/5432#issuecomment-1155667416) in [issue](https://github.com/sourcegraph/about/issues/5432#issuecomment-1156593359)). From the [MUI Docs](https://mui.com/material-ui/guides/minimizing-bundle-size/) it seems that tree-shaking should be happening automatically, and the site seems load quickly despite the larger bundle size measured on the production build. I am looking to learn more about this, so I appreciate any feedback!

### Test

1. Ensure prettier has standardized the proposed changes.
2. Check that icons are rendered according to the new style on the following pages
  - `/use-cases` and Use Case pages
  - `/code-insights`
  - `/case-studies/nutanix-fixed-log4j-with-sourcegraph`
3. Please check that there are no breaking changes with plain icons
  - Arrows in carousels and blockquotes appear and function as expected
  - Company logos (GitHub, Twitter, etc.) render at consistent sizes
